From 29fe68f2657df503926e629477a41f7d9435048f Mon Sep 17 00:00:00 2001
From: Boudhayan Gupta <me@BaloneyGeek.com>
Date: Wed, 23 Sep 2015 00:25:36 +0530
Subject: [PATCH] Fail Baloo::File::load() if the Database is not open. Fixes
 crash if selecting multiple files in Dolphin with Baloo disabled.

BUG: 353049
REVIEW: 125352
---
 src/lib/file.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/lib/file.cpp b/src/lib/file.cpp
index 6bff57d..ac034eb 100644
--- a/src/lib/file.cpp
+++ b/src/lib/file.cpp
@@ -98,6 +98,10 @@ bool File::load()
     Database *db = globalDatabaseInstance();
     db->open(Database::OpenDatabase);
 
+    if (!db->isOpen()) {
+        return false;
+    }
+
     quint64 id = filePathToId(QFile::encodeName(d->url));
     if (!id) {
         return false;
-- 
2.5.0

